chapters/overfitting.qmd

```diff
@@ -0,0 +1,487 @@
+
+sim_new %>%
+  ggplot(aes(A, B)) +
+  geom_point(
+    aes(col = class, pch = class),
+    alpha = 3 / 4,
+    cex = 2,
+    show.legend = FALSE
+  ) +
+  coord_equal() +
+  geom_contour(
+    data = grid,
+    aes(z = lp),
+    breaks = 0,
+    col = "black",
+    linewidth = 2,
+    alpha = 1 / 12
+  ) +
+  lims(x = c(-3, 3), y = c(-4, 4)) +
+  labs(x = "Predictor A", y = "Predictor B")+
+  geom_contour(
+    data = high_cost,
+    aes(z = value),
+    breaks = 0,
+    col = "black",
+    linewidth = 1,
+    alpha = 3 / 4
+  )  +
+  labs(x = "Predictor A", y = NULL) +
+  scale_color_manual(values = sim_cols)
+```
+
+Our goal is to find tuning parameter values that are _just right_: complex enough to accurately represent the data but not complex enough to over-interpret it. 	
+	
+## The Ways Models Overfit {#sec-how-to-overfit}
+
+This leads us to a formal definition of overfitting: 
+
+::: {.important-box}
+
+Overfitting is the situation where a model over-interprets trends and patterns in the training set that do not generalize to new data. 
+
+These irreproducible data trends result in models that **only do well on the training set**. 
+
+:::
+
+We can overinterpret the training set data in a few different ways. For example, @sec-imbalances will discuss class imbalance problems that occur when one or more outcome classes have a very low probability of occurrence. As models are allowed more complexity, they will often maximize accuracy (or some proxy for it) by simply declaring that all data points belong to the majority class. If there is a 1% event rate, 99% accuracy is easily achievable by overfitting to the majority class. 
+
+Another example is overfitting the predictor set. The "low $N$, high $P$" problem [@Johnstone2009] occurs when data points are expensive to obtain, but each comes with abundant information. For example, in high-dimensional biology data, it is common to have dozens of data points and hundreds of thousands of predictors. Many models cannot be estimated with this imbalance of dimensions, and often, the first step is to filter out uninformative predictors. 
+
+Suppose some statistical inferential method, such as a t-test, is used to determine if a predictor can differentiate between two classes. Small sample sizes make the chance of uninformative predictors falsely passing the filter large. If the same data are then used to build and evaluate the model, there is a significant probability that an ineffective model will appear to be very predictive [@Ambroise2002p1493;@fes]. Again, using a data set external to the model training and development process shows you when this occurs. 
+
+Finally, it is possible to overfit a model via sample filtering. For example, in some machine learning competitions, participants might sub-sample the training set to be as similar to the unlabelled test set as possible. This will most likely improve their test set performance, but it generally handicaps a model for other data sets. 
```

Comment:
```suggestion
Finally, it is possible to overfit a model via sample filtering. For example, in some machine learning competitions, participants might sub-sample the training set to be as similar to the unlabelled test set as possible. This will most likely improve their test set performance, but it generally handicaps a model when predicting a new yet-to-be-seen data set. 
```